editoast/src/views/authz.rs

```diff
@@ -83,161 +179,579 @@ async fn whoami(Extension(auth): AuthenticationExt) -> Result<Json<WhoamiRespons
+                    Resource::Infra,
+                    vec![ResourceGrant {
+                        resource_type: Resource::Infra,
+                        resource_id: *infra_id,
+                        grant: InfraGrant::Writer,
+                    }],
+                );
+            }
 
-async fn check_user_exists(user_id: i64, regulator: &Regulator) -> Result<()> {
-    if !regulator
-        .user_exists(user_id)
-        .await
-        .map_err(AuthzError::from)?
-    {
-        return Err(NoSuchUserError::NoSuchUser { user_id }.into());
+            let is_owner = authorizer
+                .check_infra_grant_owner(*infra_id)
+                .await
+                .map_err(AuthzError::from)?;
+            if is_owner {
+                result.insert(
+                    Resource::Infra,
+                    vec![ResourceGrant {
+                        resource_type: Resource::Infra,
+                        resource_id: *infra_id,
+                        grant: InfraGrant::Owner,
+                    }],
+                );
+            }
+        }
     }
 
-    Ok(())
+    Ok(Json(result))
 }
 
 #[utoipa::path(
-    get, path = "",
+    get,
+    path = "",
     tag = "authz",
-    params(UserIdPathParam),
+    params(ResourceTypeParam, ResourceIdParam, PaginationQueryParams),
     responses(
-        (status = 200, description = "List the roles of a user", body = inline(Roles)),
+        (status = 200, description = "Get list of user that have access to the resource", body = inline(Vec<SubjectGrant>)),
     ),
 )]
-async fn list_user_roles(
-    Path(UserIdPathParam { user_id }): Path<UserIdPathParam>,
-    Extension(auth): AuthenticationExt,
-    State(AppState { regulator, .. }): State<AppState>,
-) -> Result<Json<Roles>> {
-    if !auth
-        .check_roles([Role::OperationalStudies].into())
-        .await
-        .map_err(AuthorizationError::from)?
-    {
-        return Err(AuthorizationError::Forbidden.into());
+async fn users_grants_for_resource_id(
+    State(AppState {
+        regulator, db_pool, ..
+    }): State<AppState>,
+    Path(resource_type_param): Path<ResourceTypeParam>,
+    Path(resource_id_param): Path<ResourceIdParam>,
+    Query(pagination_params): Query<PaginationQueryParams>,
+) -> Result<Json<Vec<SubjectGrant>>> {
+    // Checking if the resource type is valid
+    // For now we only support infra
+    if resource_type_param.resource_type != Resource::Infra {
+        return Err(ResourceError::ResourceTypeNotFound {
+            resource_type: resource_type_param.resource_type.to_string(),
+        })?;
+    }
+
+    // Check that infra exists
+    let infra_id = resource_id_param.resource_id;
+    let conn = &mut db_pool.get().await?;
+    Infra::retrieve_or_fail(conn, infra_id, || InfraApiError::NotFound { infra_id }).await?;
+
+    // Validate pagination params
+    let (page, page_size) = pagination_params
+        .validate(1000)?
+        .warn_page_size(100)
```

Comment:
```suggestion
        .validate(100)?
```

the warn is only there for API retrocompatibility